sui-execution/latest/sui-adapter/src/programmable_transactions/linkage_view.rs

```diff
@@ -281,6 +267,36 @@ impl<'state> LinkageResolver for LinkageView<'state> {
     }
 }
 
+impl From<&MovePackage> for LinkageInfo {
+    fn from(package: &MovePackage) -> Self {
+        Self {
+            storage_id: package.id().into(),
+            runtime_id: package.original_package_id().into(),
+            link_table: package.linkage_table().clone(),
+        }
+    }
+}
+
+impl<'state> LinkageResolver for LinkageView<'state> {
```

Comment:
I think this is not needed any longer strictly speaking and it was needed when `DataStore` was implemented here.
However I have the feeling it will help in the future to have the implementation of those function in the `impl` for `LinkageView` so I'm leaving the change in.
Let me know if you are bothered too much by this